model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -29,52 +35,115 @@ class CTCGreedyDecoderReplacement(FrontReplacementSubgraph):
     CTCGreedyDecoder and SparseToDense operations and removes the SparseToDense and Cast operation which is also used
     in the SparseToDense operation, because Inference Engine implementation of the CTCGreedyDecoder produces regular
     tensor as output.
-
-    The second input to the CTCGreedyDecoder in the TensorFlow is a 1D tensor with sequence lengths. In the Inference
-    Engine the second input to the CTCGreedyDecoder is a 2D tensor where the first element in each row is equal to 0
-    and all others are equal to 1. The length of the row is equal to the sequence length. The replacer modifies the
-    second input to be compatible with the Inference Engine CTCGreedyDecoder layer implementation.
+    Also, Inference Engine CTCGreedyDecoder requires a mask format for sequence lengths that is a different from
+    original one. Hence, this transformation changes a format of sequence length to a mask by replacing Fill and Pack
+    nodes with a special graph that produces a tensor of ones with shape [T, N] accepted by opset CTCGreedyDecoder.
     """
     enabled = True
 
+    def run_before(self):
+        return [Pack, FillToBroadcast]
+
     @staticmethod
     def pattern(**kwargs):
         return dict(
             nodes=[
+                ('transpose', dict(op='Transpose')),
+                ('shape', dict(op='ShapeOf')),
+                ('shape_1', dict(op='ShapeOf')),
+                ('strided_slice', dict(op='StridedSlice')),
+                ('stack', dict(op='Const', value=[1])),
```

Comment:
it is not safe to match node with values like that. we should check like 
`value=lambda v: v is not None and np.array_equal(v, [1])`

Reply:
Ok, will fix it.

Reply:
Fixed.